src/shared/curl-util.h

```diff
@@ -16,6 +16,7 @@ typedef struct CurlGlue {
 
         void (*on_finished)(CurlGlue *g, CURL *curl, CURLcode code);
         void *userdata;
+        struct curl_slist *resolve_rules;;
```

Comment:
so if you move curl-util.[ch] into src/shared/ (which is acceptable) you also have to dlopen()-ify it, see how this is done for other libs, i.e. grep for DLSYM_ARG and related stuff.

i.e. we shouldn't be pulling libcurl into the minimal dep set.